include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -1341,34 +1341,36 @@ namespace LinearAlgebra
 #ifdef DEAL_II_WITH_MPI
 
 #  ifdef DEBUG
-      if (Utilities::MPI::job_supports_mpi())
+      Assert(Utilities::MPI::job_supports_mpi() ||
+               (update_ghost_values_requests.empty() &&
+                compress_requests.empty()),
+             ExcInternalError());
+
+      // make sure that there are not outstanding requests from updating
+      // ghost values or compress
+      int flag = 1;
```

Comment:
While you're there, I think you can push the declaration of this variable into each of the following two scopes and make it more local.